webapp/channels/src/utils/url.tsx

```diff
@@ -184,6 +184,7 @@ export function shouldOpenInNewTab(url: string, siteURL?: string, managedResourc
         // Paths managed by plugins and public file links aren't handled by the web app
         'plugins',
         'files',
+        'api/v4/reports/export',
```

Comment:
I've abandoned the idea of having a separate post type and just decided to use the default post type, and instead tell the markdown processor to open links from the API in a new window.

Reply:
Simpler the better.

Reply:
Yes that's what the above code does, it makes an exception for internal links that start with the above paths.

Reply:
Hmm, I was thinking why we didn't just attach the file as a regular attachment, then I went through Jesse's comments in the earlier PR. I'm not sure if its worth it IMO. We didn't do the file attachment because the admin may want to share the file and can't do it with download and sharing with whoever they want to with. Is the complexity of modifying markdown code in web, plus mobile, worth it? IMO not, but your call.

What would happen if I open the link from desktop app? Does it open in a new window? If so, does the window close automatically? If it does, how do we decide if to, and when to auto-close it. If we don';t auto-close it, then admin is anyways doing a manunal step, then let them just download and share manually. If we open in browser instead of desktop app window, then user may not be logged in browser.

IMHO, I'm 3/5 we should do it as a regular file attachment and not this dedicated-api approach.

Reply:
We don't know if, and what plugin we'd break with this change. Have we considered the downside of this approach?

Reply:
I tried it from the latest desktop app (v5.6.0) and the file link opened up in system's default browser, which is incorrect because the user may not be logged in there. As was the case with me, I got the unauthenticated error on opening the link-

```json
{
	"id": "api.context.session_expired.app_error",
	"message": "Invalid or expired session, please login again.",
	"detailed_error": "token=bb46rd7d9bfgtdr658ny6ebs3o",
	"request_id": "wj81i9dkkjrg9dqp9xi6n17x6c",
	"status_code": 401
}
```

Reply:
So unfortunately the Desktop App case does seem to break this flow (you'd think I'd noticed that :P) as the `target=_blank` link embedded in the post does lead outside the browser. While I'll maintain I think the importance of being able to share the report, we can't have Desktop App users forced into an error screen every single time. 

So I've changed this to use a file attachment as we had originally designed, and removed the API endpoint for retrieval.

As an aside note, I did review the risk wrt. the open link in new tab change and it's pretty benign, since all it would do is force a new tab for certain links, and shouldn't affect plugins at all since they're on a new route.

Reply:
Hopefully we could allow this sort of behaviour with the desktop app changes we've wanted to do. I notice that plugins are currently able to do it, but I think that's mostly for Boards/Playbooks because they have their own special behaviour anyway of being able to open more tabs